pkg/model/config/config.go

```diff
@@ -147,41 +153,43 @@ func (c Config) resourceAPIVersionCompatible(verType, version string) bool {
 	return true
 }
 
-// GVK contains information about scaffolded resources
-type GVK struct {
-	Group   string `json:"group,omitempty"`
-	Version string `json:"version,omitempty"`
-	Kind    string `json:"kind,omitempty"`
-
-	// CRDVersion holds the CustomResourceDefinition API version used for the GVK.
-	CRDVersion string `json:"crdVersion,omitempty"`
-	// WebhookVersion holds the {Validating,Mutating}WebhookConfiguration API version used for the GVK.
-	WebhookVersion string `json:"webhookVersion,omitempty"`
-}
-
-// isEqualTo compares it with another resource
-func (r GVK) isEqualTo(other GVK) bool {
-	return r.Group == other.Group &&
-		r.Version == other.Version &&
-		r.Kind == other.Kind
-}
-
-// merge combines fields of two GVKs that have matching group, version, and kind,
-// favoring the receiver's values.
-func (r *GVK) merge(other GVK) {
-	if r.CRDVersion == "" && other.CRDVersion != "" {
-		r.CRDVersion = other.CRDVersion
-	}
-	if r.WebhookVersion == "" && other.WebhookVersion != "" {
-		r.WebhookVersion = other.WebhookVersion
-	}
-}
-
 // Marshal returns the bytes of c.
 func (c Config) Marshal() ([]byte, error) {
 	// Ignore extra fields at first.
 	cfg := c
 	cfg.Plugins = nil
+
+	// Ignore some fields if v2.
+	if cfg.IsV2() {
+		for i := range cfg.Resources {
+			cfg.Resources[i].Domain = ""
+			cfg.Resources[i].Plural = ""
+			cfg.Resources[i].Path = ""
+			cfg.Resources[i].API = nil
+			cfg.Resources[i].Controller = false
+			cfg.Resources[i].Webhooks = nil
+		}
+	}
+
+	for i, r := range cfg.Resources {
```

Comment:
Is not it only required for v2? I mean, could we add `if cfg.IsV2() {` here too?